insights/core/spec_cleaner.py

```diff
@@ -336,35 +345,54 @@ def _redact_line(self, line):
         # redact line per the file-content-redaction.yaml
         if not line:
             return line
-        # patterns removal
+        # 1. patterns removal
         find = re.search if self.redact['regex'] else lambda x, y: x in y
         if any(find(pat, line) for pat in self.redact.get('exclude', [])):
             logger.debug("Pattern matched, removing line: %s" % line.strip())
             # patterns found, remove it
             return None
-        # password removal
+        # 2. password removal
         for regex in DEFAULT_PASSWORD_REGEXS:
             tmp_line = line
             line = re.sub(regex, r"\1\2********", tmp_line)
             if line != tmp_line:
                 break
-        # keyword replacement redaction
+        # 3. keyword replacement redaction
         return self._sub_keywords(line)
 
+    def get_obfuscate_functions(self, _file, no_obfuscate=None):
+        obf_funcs = []
+        # Get the actual obfuscate list setting for this file
+        obfs = set(self.obfuscate) - set(no_obfuscate or [])
+        # IP obfuscation entry
+        obf_funcs.append(self._sub_ip_netstat if _file.endswith("netstat_-neopa") else self._sub_ip) if "ip" in obfs else None
+        # Hostname obfuscation entry
+        obf_funcs.append(self._sub_hostname) if "hostname" in obfs else None
+        return obf_funcs
+
     def clean_content(self, lines, filters=None, obf_funcs=None, no_redact=False):
         """
         Clean lines one by one according to the configuration, the cleaned
         lines will be returned.
         """
+        def _clean_line(_line):
+            # 1. Do Redaction by default, unless "no_redact=True"
+            if _line and not no_redact:
+                _line = self._redact_line(_line)
+            # 2. Do Filtering as per the "filters"
+            _line = self._filter_line_allowlist(_line, filters)
+            # 3. Do Obfuscation as per the "obf_funcs"
+            _line = self._obfuscate_line(_line, obf_funcs or [])
+            return _line
+
+        # handle single string
+        if not isinstance(lines, list):
+            return _clean_line(lines)
+
+        filters = filters or []
         result = []
         for line in lines:
-            # 1. Do Redaction by default, unless 'no_redact=True'
-            if line and not no_redact:
-                line = self._redact_line(line)
-            # 2. TODO: Do Filtering as per the "filters"
-            # line = self._filter_line_allow(line, filters)
-            # 3. Do Obfuscation as per the "obf_funcs"
-            line = self._obfuscate_line(line, obf_funcs or [])
+            line = _clean_line(line)
             result.append(line) if line is not None else None
```

Comment:
This `if line is not None` checking could introduce an interesting corner case: 
   - For a collected content, after cleaner.clean_content(), it has the
`result`  `['\n', '\n']`. 

Should this be collected as the final content of the file?     
Or this file should be emptied and skipped in insights-archive?  

See the test-case example in https://github.com/RedHatInsights/insights-core/pull/4061/files .

Reply:
Per our discussion in yesterday's sync, I opened a new card to track the excluding of the empty files RHINENG-8978